adele-ticket-master/src/main/java/com/microservicesteam/adele/ticketmaster/TicketMasterService.java

```diff
@@ -30,36 +32,92 @@
 
     @Subscribe
     public void handleCommand(CreateTickets command) {
-        addTickets().accept(command);
-        eventBus.post(TicketsCreated.builder()
-                .addAllPositions(command.positions())
-                .build());
+        if (ticketsNotExist(command)) {
+            addTickets(command);
+            eventBus.post(TicketsCreated.builder()
+                    .addAllPositions(command.positions())
+                    .build());
+        } else {
+            eventBus.post(NoOperation.builder()
+                    .sourceCommand(command)
+                    .build());
+        }
     }
 
+
     @Subscribe
     public void handleCommand(BookTickets command) {
-        bookTickets().accept(command);
-        eventBus.post(TicketsBooked.builder()
-                .bookingId(command.bookingId())
-                .addAllPositions(command.positions())
-                .build());
+        if (allTicketsFree(command)) {
+            bookTickets(command);
+            eventBus.post(TicketsBooked.builder()
+                    .bookingId(command.bookingId())
+                    .addAllPositions(command.positions())
+                    .build());
+        } else {
+            eventBus.post(NoOperation.builder()
+                    .sourceCommand(command)
+                    .build());
+        }
+    }
+
+    @Subscribe
+    public void handleCommand(CancelTickets command) {
+        if (allTicketsBooked(command)) {
+            cancelTickets(command);
+            eventBus.post(TicketsCancelled.builder()
+                    .bookingId(command.bookingId())
+                    .addAllPositions(command.positions())
+                    .build());
+        } else {
+            eventBus.post(NoOperation.builder()
+                    .sourceCommand(command)
+                    .build());
+        }
+    }
+
+    private boolean ticketsNotExist(CreateTickets command) {
```

Comment:
Maybe it's just me, but feels weird that I can only check if tickets exist if I have a `CreateTickets` instance. What is the reason behind this?
This applies to the methods (`allTicketsFree`, `allTicketsBooked`, etc) below as well.